packages/integration_test/lib/integration_test.dart

```diff
@@ -191,4 +194,79 @@ class IntegrationTestWidgetsFlutterBinding
     );
     results[description] ??= _success;
   }
+
+  vm.VmService _vmService;
+
+  /// Initialize the [vm.VmService] settings for the timeline.
+  @visibleForTesting
+  Future<void> enableTimeline({
+    List<String> streams = const <String>['all'],
+    @visibleForTesting vm.VmService vmService,
+  }) async {
+    assert(streams != null);
+    assert(streams.isNotEmpty);
+    if (vmService != null) {
+      _vmService = vmService;
+    }
+    if (_vmService == null) {
+      final developer.ServiceProtocolInfo info =
+          await developer.Service.getInfo();
+      assert(info.serverUri != null);
+      _vmService = await vm_io.vmServiceConnectUri(
+        'ws://localhost:${info.serverUri.port}${info.serverUri.path}ws',
+      );
+    }
+    await _vmService.setVMTimelineFlags(streams);
+  }
+
+  /// Runs [action] and outputs a [vm.Timeline] trace for it.
+  ///
+  /// Waits for the `Future` returned by [action] to complete prior to stopping
+  /// the trace.
+  ///
+  /// `streams` limits the recorded timeline event streams to only the ones
+  /// listed. By default, all streams are recorded.
+  /// See `timeline_streams` in
+  /// https://github.com/dart-lang/sdk/blob/master/runtime/vm/timeline.cc
+  ///
+  /// If [retainPriorEvents] is true, retains events recorded prior to calling
+  /// [action]. Otherwise, prior events are cleared before calling [action]. By
+  /// default, prior events are cleared.
+  Future<vm.Timeline> traceTimeline(
+    Future<dynamic> action(), {
+    List<String> streams = const <String>['all'],
+    bool retainPriorEvents = false,
+  }) async {
+    await enableTimeline(streams: streams);
+    if (retainPriorEvents) {
+      await action();
+      return await _vmService.getVMTimeline();
+    }
+
+    await _vmService.clearVMTimeline();
+    final vm.Timestamp startTime = await _vmService.getVMTimelineMicros();
+    await action();
+    final vm.Timestamp endTime = await _vmService.getVMTimelineMicros();
+    return await _vmService.getVMTimeline(
+      timeOriginMicros: startTime.timestamp,
+      timeExtentMicros: endTime.timestamp,
+    );
+  }
+
+  /// This is a convience wrap of [traceTimeline] and send the result back to
+  /// the host.
```

Comment:
>  This is a convenience wrapper of [traceTimeline] that sends the result back to the host.

We should explain a bit more about how - that this only works with driver style tests using the `reportData` member on the binding.

Reply:
We should also document the use of the reportKey parameter here, and that callers should be sure to make this a unique key for the reportData in the test suite.

For other parameters it's probably fine to just say they are passed through as-is to [traceTimeline].

Reply:
Done.